### PR TITLE
Fail service unbind if bound to apps and `--unbind` not specified. New option.

### PR DESCRIPTION
**Pull request by andreas-kupries**
Fix: #1359 

:construction: Will not pass CI. Tests not adapted yet.

### CODE REVIEW

File: internal/api/v1/service/delete.go [hunk 1]
```diff
@@ -23,6 +48,76 @@ func (ctr Controller) Delete(c *gin.Context) apierror.APIErrors {
...
+		if errors.Is(err, helmdriver.ErrReleaseNotFound) {
+			return apierror.NewNotFoundError(fmt.Sprintf("%s - %s", err.Error(), releaseName))
+		}
+		return apierror.InternalError(err)
+	}
+
+	logger.Info(fmt.Sprintf("service found %+v\n", serviceName))
+	if srv.Info.Status != helmrelease.StatusDeployed {
+		return apierror.InternalError(err)
+	}
```
Comment (andreas-kupries) [inline internal/api/v1/service/delete.go:71]: @enrichman I added this block of code to have the `delete service` endpoint check that the instance to be deleted actually exists.

As is done by `bind` and `unbind`. After adapting the delete tests to provide the new delete request structure I consistently get `404 release not found` coming from this code.

Was it intentional to not have this validation in the delete endpoint ?
If not then I will have to dig again into the test setup to see if/why no release is present ...
And/or test cli first, as that should match on all sides.

(side note: should be factored into a helper function)

Comment (andreas-kupries) [reply internal/api/v1/service/delete.go:71]: Yes, it is definitely this piece of code tripping the test now.

**Comment by andreas-kupries** [comment]
Big rebase, and insertion of fixes.

The big issue was the truncation of the name for the service helm chart CR.

The way we do that it actually expands the name to the given limit. And then comes the chart controller and derives the name of the release from that, which in turn derives pod names etc. from that. And neither seems to perform proper truncation.
Thus we have to make the name of the helm hart CR short enough to leave space for all these to append their stuff without running the kube limits. Under the length of the shasum. Made that possible now, forced a leading letter. Believing that the chance of collision is still low enough.

With this done everything creating a service as part of a test should now work instead of timing out because the service was not deployed without any error messages.

Still leaves issues with possibly broken tests.

**Comment by andreas-kupries** [comment]
The -cli failure
> In [AfterEach] at: /home/sles/actions-runner/_work/epinio/epinio/acceptance/services_test.go:98
is for 
> /home/sles/actions-runner/_work/epinio/epinio/acceptance/services_test.go:103
The same test passes locally.

**Comment by andreas-kupries** [comment]
-api failure

  - app stage is likely unrelated
  - service show I have to look into - something in the setup
  - service delete - @enrichman I hope you can look at that one. It should fail locally also. And I am unclear why. There is something in that test I am not understanding.

**Comment by andreas-kupries** [comment]
Ah. Service Show is because I added `wait for helm release` to `CreateHelmChart`. And in that test it has to fail.
Ok. Make waiting conditional, and disable for that one case.

**Comment by andreas-kupries** [comment]
Yay. A single failing test.

>  STEP: creating a namespace: namespace-849391446 05/17/22 09:49:04.849
    STEP: targeting a namespace: namespace-849391446 05/17/22 09:49:23.86
    STEP: MSI service-428751036 -> service-instance-787772250 05/17/22 09:49:25.788
    STEP: MSI/ok 05/17/22 09:49:30.695
    STEP: locate helm chart x90de42de07a57973259f04cc2b9e 05/17/22 09:49:30.695
    STEP: assemble url 05/17/22 09:49:30.817
    STEP: assemble request for https://epinio.192.168.96.4.omg.howdoi.website/api/v1/namespaces/namespace-849391446/services/service-instance-787772250 05/17/22 09:49:30.817
    STEP: curl request 05/17/22 09:49:30.817
    STEP: read response 05/17/22 09:49:33.078
    STEP: decode response {"errors":[{"status":404,"title":"release: not found - x90de42de07a57973259f04cc2b9e","details":""}]} 05/17/22 09:49:33.078
    STEP: check status 05/17/22 09:49:33.078
    STEP: deleting a namespace: namespace-849391446 05/17/22 09:49:33.627
  << End Captured GinkgoWriter Output

>  {"errors":[{"status":404,"title":"release: not found - x90de42de07a57973259f04cc2b9e","details":""}]}
  Expected
      <int>: 404
  to equal
      <int>: 200
  In [It] at: /home/sles/actions-runner/_work/epinio/epinio/acceptance/api/v1/service_delete_test.go:157

It is the one where I do not understand why it fails.
A service (instance) is created.
When deleting that service the validation of the request finds the service, yet not the helm release for it.
Thus is reports the `404 release not found`.
Yet the setup should have created the release, even waited for it to be deployed, IIUC.

@enrichman ^ Please have a look.

**Review by enrichman — APPROVED** [review]

**Pull request merged**